From 4f6d61a1cd54d05dd7f29e9921e08a4f27ba0e1c Mon Sep 17 00:00:00 2001
From: Mikko Perttunen <mperttunen@nvidia.com>
Date: Fri, 19 May 2017 10:48:54 +0300
Subject: [PATCH 333/337] rwsem_rt: fix compilation error on -Werror

Comparison between signed and unsigned variables causes compilation
error on some compilers when -Werror is enabled.

Bug 1905220

Change-Id: I8ed181add09c0cadce77bfdecfc5b7cb87f33730
Signed-off-by: Mikko Perttunen <mperttunen@nvidia.com>
---
 include/linux/rwsem_rt.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/rwsem_rt.h b/include/linux/rwsem_rt.h
index 2ffbf09..2aa90f3 100644
--- a/include/linux/rwsem_rt.h
+++ b/include/linux/rwsem_rt.h
@@ -47,7 +47,7 @@ do {							\
 
 static inline int rwsem_is_locked(struct rw_semaphore *sem)
 {
-	return atomic_read(&sem->readers) != READER_BIAS;
+	return atomic_read(&sem->readers) != (int)READER_BIAS;
 }
 
 static inline int rwsem_is_contended(struct rw_semaphore *sem)
-- 
2.1.4

